### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by awais-amjed**
In this Pull Request, the following changes were introduced:

- Created a Person Class. ✅ 
- Added Student and Teacher classes inheriting from Person. ✅ 
- Add professional readme. ✅

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
```
Comment (ShoiraTa) [inline person.rb]: - [ ] Kindly add `@id` inside constructor and give it random number, eg.:
`@id = Random.rand(1..1000)`

- [ ] I strongly recommend starting the page from getters and setters to follow best practices: 

```ruby
class Person
  attr_accessor :name, :age 
  attr_reader :id 
 
  def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age?
+  end
```
Comment (ShoiraTa) [inline person.rb]: - [ ] Public method can_use_services? that returns true if person is of age or if they have permission from parents. in your case it is only the age. Lets add parent permission 😉 
```ruby
  def can_use_services? 
    of_age?() || @parent_permission
  end
```

- [ ] OPTIONAL: Usually private methods come at the end 😉

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Student < Person
+  def initialize(age, classroom, name = nil, parent_permission: nil)
```
Comment (ShoiraTa) [inline student.rb]: - [ ] parent_permission by default should be true.
- [ ] name by default should be unknown

`def initialize(age, classroom, name = "Unknown", parent_permission = true)`

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Teacher < Person
+  def initialize(age, specialization, name = nil, parent_permission = nil)
```
Comment (ShoiraTa) [inline teacher.rb]: - [ ] parent_permission by default should be `true`.
- [ ] name by default should be `unknown`

`def initialize(age, specialization, name = "Unknown", parent_permission = true)`

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @awais-amjed ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Awais, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**